doc/deployment_guide.md

```diff
@@ -0,0 +1,806 @@
+Oct 28 16:37:17 mdip-gatekeeper start-node[706734]: tbtc-mediator-1  |     "complete": true
+Oct 28 16:37:17 mdip-gatekeeper start-node[706734]: tbtc-mediator-1  | }
+Oct 28 16:37:17 mdip-gatekeeper start-node[706734]: tbtc-mediator-1  | 0.00017858000000000001
+Oct 28 16:37:17 mdip-gatekeeper start-node[706734]: tbtc-node-1      | 2024-10-28T19:43:17.461639Z [mdip] AddToWallet 4b52dfa95d7b9472147b2821a33c59cdd52bf1444b6c0feed4e7c8c0d9a6f798  new InMempool
+Oct 28 16:37:17 mdip-gatekeeper start-node[706734]: tbtc-mediator-1  | Transaction broadcast with txid: 4b52dfa95d7b9472147b2821a33c59cdd52bf1444b6c0feed4e7c8c0d9a6f798
+Oct 28 16:37:18 mdip-gatekeeper start-node[706734]: gatekeeper-1     | POST /api/v1/queue/TBTC/clear 200 650.769 ms - 4
+Oct 28 16:37:18 mdip-gatekeeper start-node[706734]: tbtc-mediator-1  | export loop waiting 1 minute(s)...
+```
+
+## Secure Hosting an MDIP Node with nginx
+
+nginx [https://nginx.org/](https://nginx.org/) is a popular web server and reverse proxy available for most operating systems.
+
+nginx can be used to control and throttle access to server ports and hosted services like MDIP. The configuration files presented here are for educational purposes. Securing a server for online operations involves more than protecting a few ports; this document limits itself to MDIP-specific ports and services.
+
+### Important MDIP Ports and Endpoints
+
+A full MDIP node launched using the provided `kc/docker-compose.yml` configuration will only expose 2 new service ports on the Host machine:
+
+```docker
+keychainmdip/keymaster   0.0.0.0:4226->4226/tcp   kc-keymaster-1
+keychainmdip/gatekeeper  0.0.0.0:4224->4224/tcp   kc-gatekeeper-1
+```
+
+When using Docker as an operating environment, other ports are used within the Docker network to mediate with 3rd party systems.
+
+```docker
+mongo:latest             27017/tcp kc-mongodb-1
+bitcoin-core:v27.99.0    48332/tcp kc-tbtc-node-1
+feathercoind             19337/tcp kc-tftc-node-1
+```
+
+None of the ports listed above should ever be directly addressable from the Internet. It is recommended that a firewall be setup to prevent access to these services.
+
+### Basic nginx Hosting
+
+The nginx configuration file below can be used to launch a live MDIP node that exposes only read-only gatekeeper end-points.
+
+```nginx
+#
+# Basic read-only MDIP Gatekeeper server routes configuration for nginx
+#
+server {
+    server_name mdip.mydomain.net; # Replace with your domain name
+    access_log /var/log/mdip-node; 
```

Comment:
Consider moving to `/var/log/nginx/mdip-node.log`, both for directory and file extension standardization.

Reply:
great recommendations. work tbd before I update the guide.